### ISSUE TITLE
RedirectException causing TypeError

### ISSUE THREAD

**Issue by malcolmwhite** [4Catalyzer/found-relay#44]
I just posted this to SO -- sorry for the issue noise as there's a good chance it is user error.

I'm trying to redirect unauthenticated users to a login page by throwing a RedirectException. To get the ball rolling, I'm using the `relay-todomvc-found-modern` demo and trying to redirect the user to a login page if the status variable is active. The redirect works if I visit `/completed` and click the link for `/active`, but throws `TypeError: Cannot read property 'getHostNode' of null` if I directly visit `/active`.

Starting with the relay-todomvc-found-modern demo, I added the following lines:

```
import { RedirectException } from 'found';

...
render() {
     if (status === 'active') {
         throw new RedirectException({pathname: '/login'})
     }
}
...
```

where `/login` leads to a component that is copy/pasted from `TodoList` (with name changes as necessary) that is set up with the following route:

```
<Route
  path="/"

...

    <Route
        path="login"
        Component={Login}
        query={LoginQuery}
        prepareVariables={params => ({ ...params, status: 'any' })}
    />

...

</Route>
```

And the `LoginQuery` is

```
const LoginQuery = graphql`
    query routes_Login_Query($status: String!) {
        viewer {
            ...Login_viewer
        }
    }
`;
```
The `Login` component is exactly the same as the `TodoList` components, except without the `RedirectException` and with name changes as necessary.

**Comment by taion** [comment]
You need to throw it in `prerender` (for now, might get changed by https://github.com/4Catalyzer/found-relay/issues/43 at some point).

See https://github.com/4Catalyzer/found-relay/blob/master/test/prerender.test.js.

**Comment by taion** [comment]
Would be resolved by #45.

**Comment by malcolmwhite** [comment]
Thanks for the quick response!
